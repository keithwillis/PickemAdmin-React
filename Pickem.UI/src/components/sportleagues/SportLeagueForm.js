import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import CheckBoxInput from "../common/CheckBoxInput";
import ConferenceList from "../conferences/ConferenceList";

const SportLeagueForm = ({
  sportLeague,
  conferences,
  divisions,
  teams,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <>
      <form onSubmit={onSave}>
        <h2>{sportLeague.id ? "Edit" : "Add"} Sport League</h2>
        {errors.onSave && (
          <div className="alert alert-danger" role="alert">
            {errors.onSave}
          </div>
        )}
        <TextInput
          name="name"
          label="Name"
          value={sportLeague.name}
          onChange={onChange}
          error={errors.name}
        />
        <CheckBoxInput
          name="isEnabled"
          label="Is Enabled"
          checked={sportLeague.isEnabled}
          value="IsEnabled"
          onChange={onChange}
          error={errors.isEnabled}
        />
        <button type="submit" disabled={saving} className="btn btn-primary">
          {saving ? "Saving..." : "Save"}
        </button>
      </form>
      <br />
      <br />

      <ConferenceList
        conferences={conferences}
        divisions={divisions}
        teams={teams}
        sportLeagueId={sportLeague.id}
      ></ConferenceList>
    </>
  );
};

SportLeagueForm.propTypes = {
  conferences: PropTypes.array,
  divisions: PropTypes.array,
  teams: PropTypes.array,
  sportLeague: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool
};

export default SportLeagueForm;
