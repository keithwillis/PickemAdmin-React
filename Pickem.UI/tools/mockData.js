const teams = [
  {
    id: 1,
    name: "Ravens",
    location: "Baltimore",
    isActive: true,
    divisionId: 1
  },
  {
    id: 2,
    name: "Seahawks",
    location: "Seattle",
    isActive: true,
    divisionId: 8
  },
  {
    id: 3,
    name: "49ers",
    location: "San Francisco",
    isActive: true,
    divisionId: 8
  },
  {
    id: 4,
    name: "Giants",
    location: "New York",
    isActive: true,
    divisionId: 7
  },
  {
    id: 5,
    name: "Redskins",
    location: "Washington",
    isActive: true,
    divisionId: 7
  },
  {
    id: 6,
    name: "Cowboys",
    location: "Dallas",
    isActive: true,
    divisionId: 7
  },
  {
    id: 7,
    name: "Eagles",
    location: "Philadelphia",
    isActive: true,
    divisionId: 7
  },
  {
    id: 8,
    name: "Panthers",
    location: "Carolina",
    isActive: true,
    divisionId: 6
  },
  {
    id: 9,
    name: "Buccaneers",
    location: "Tampa Bay",
    isActive: true,
    divisionId: 6
  },
  {
    id: 10,
    name: "Falcons",
    location: "Atlanta",
    isActive: true,
    divisionId: 6
  },
  {
    id: 11,
    name: "Saints",
    location: "New Orleans",
    isActive: true,
    divisionId: 6
  },
  {
    id: 12,
    name: "Vikings",
    location: "Minnesota",
    isActive: true,
    divisionId: 5
  },
  {
    id: 13,
    name: "Lions",
    location: "Detroit",
    isActive: true,
    divisionId: 5
  },
  {
    id: 14,
    name: "Packers",
    location: "Green Bay",
    isActive: true,
    divisionId: 5
  },
  {
    id: 15,
    name: "Bears",
    location: "Chicago",
    isActive: true,
    divisionId: 5
  },
  {
    id: 16,
    name: "Chargers",
    location: "Los Angeles",
    isActive: true,
    divisionId: 4
  },
  {
    id: 17,
    name: "Raiders",
    location: "Los Vegas",
    isActive: true,
    divisionId: 4
  },
  {
    id: 18,
    name: "Broncos",
    location: "Baltimore",
    isActive: true,
    divisionId: 4
  },
  {
    id: 19,
    name: "Chiefs",
    location: "Baltimore",
    isActive: true,
    divisionId: 4
  },
  {
    id: 20,
    name: "Dolphins",
    location: "Miami",
    isActive: true,
    divisionId: 3
  },
  {
    id: 21,
    name: "Jets",
    location: "New York",
    isActive: true,
    divisionId: 3
  },
  {
    id: 22,
    name: "Bills",
    location: "Buffalo",
    isActive: true,
    divisionId: 3
  },
  {
    id: 23,
    name: "Patriots",
    location: "New England",
    isActive: true,
    divisionId: 3
  },
  {
    id: 24,
    name: "Jaguars",
    location: "Jacksonville",
    isActive: true,
    divisionId: 2
  },
  {
    id: 25,
    name: "Colts",
    location: "Indianapolis",
    isActive: true,
    divisionId: 2
  },
  {
    id: 26,
    name: "Titans",
    location: "Tennessee",
    isActive: true,
    divisionId: 2
  },
  {
    id: 27,
    name: "Texans",
    location: "Houston",
    isActive: true,
    divisionId: 2
  },
  {
    id: 28,
    name: "Bengals",
    location: "Cincinnati",
    isActive: true,
    divisionId: 1
  },
  {
    id: 29,
    name: "Browns",
    location: "Cleveland",
    isActive: true,
    divisionId: 1
  },
  {
    id: 30,
    name: "Steelers",
    location: "Pittsburgh",
    isActive: true,
    divisionId: 1
  },
  {
    id: 31,
    name: "Rams",
    location: "Los Angeles",
    isActive: true,
    divisionId: 8
  },
  {
    id: 32,
    name: "Cardinals",
    location: "Arizona",
    isActive: true,
    divisionId: 8
  }
];

const sportLeagues = [
  {
    id: 1,
    name: "NFL",
    isEnabled: true
  },
  {
    id: 2,
    name: "NBA",
    isEnabled: false
  },
  {
    id: 3,
    name: "NHL",
    isEnabled: false
  },
  {
    id: 4,
    name: "MLB",
    isEnabled: false
  },
  {
    id: 5,
    name: "XFL",
    isEnabled: false
  },
  {
    id: 6,
    name: "MLS",
    isEnabled: false
  }
];

const conferences = [
  {
    id: 1,
    divisions: null,
    sportLeagueId: 1,
    name: "AFC",
    isActive: true
  },
  {
    id: 2,
    divisions: null,
    sportLeagueId: 1,
    name: "NFC",
    isActive: true
  }
];

const divisions = [
  {
    id: 1,
    name: "North",
    isActive: true,
    teams: null,
    conferenceId: 1
  },
  {
    id: 2,
    name: "South",
    isActive: true,
    teams: null,
    conferenceId: 1
  },
  {
    id: 3,
    name: "East",
    isActive: true,
    teams: null,
    conferenceId: 1
  },
  {
    id: 4,
    name: "West",
    isActive: true,
    teams: null,
    conferenceId: 1
  },
  {
    id: 5,
    name: "North",
    isActive: true,
    teams: null,
    conferenceId: 2
  },
  {
    id: 6,
    name: "South",
    isActive: true,
    teams: null,
    conferenceId: 2
  },
  {
    id: 7,
    name: "East",
    isActive: true,
    teams: null,
    conferenceId: 2
  },
  {
    id: 8,
    name: "West",
    isActive: true,
    teams: null,
    conferenceId: 2
  }
];

module.exports = {
  sportLeagues,
  teams,
  conferences,
  divisions
};
