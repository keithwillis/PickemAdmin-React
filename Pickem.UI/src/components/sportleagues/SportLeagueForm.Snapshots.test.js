import React from "react";
import SportLeagueForm from "./SportLeagueForm";
import renderer from "react-test-renderer";
import { sportLeagues } from "../../../tools/mockData";

it("sets submit button label 'Saving...' wen saving is true", () => {
  const tree = renderer.create(
    <SportLeagueForm
      sportLeague={sportLeagues[0]}
      conferences={[]}
      onSave={jest.fn()}
      onChange={jest.fn}
      saving
    />
  );
  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' wen saving is false", () => {
  const tree = renderer.create(
    <SportLeagueForm
      sportLeague={sportLeagues[0]}
      conferences={[]}
      onSave={jest.fn()}
      onChange={jest.fn}
      saving={false}
    />
  );
  expect(tree).toMatchSnapshot();
});
