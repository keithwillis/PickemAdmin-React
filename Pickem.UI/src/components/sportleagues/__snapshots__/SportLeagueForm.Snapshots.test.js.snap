// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' wen saving is false 1`] = `
Array [
  <form
    onSubmit={[MockFunction]}
  >
    <h2>
      Edit
       Sport League
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="name"
          onChange={[Function]}
          type="text"
          value="NFL"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="isEnabled"
      >
        Is Enabled
      </label>
      <div
        className="field"
      >
        <input
          checked={true}
          className="form-control"
          name="isEnabled"
          onChange={[Function]}
          type="checkbox"
          value="IsEnabled"
        />
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>,
  <br />,
  <br />,
  <div
    className="row"
  >
    <div
      className="col-sm-3"
    >
      <div
        className="flex-column nav nav-pills"
        onKeyDown={[Function]}
        role="tablist"
      />
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="tab-content"
      />
    </div>
  </div>,
]
`;

exports[`sets submit button label 'Saving...' wen saving is true 1`] = `
Array [
  <form
    onSubmit={[MockFunction]}
  >
    <h2>
      Edit
       Sport League
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <div
        className="field"
      >
        <input
          className="form-control"
          name="name"
          onChange={[Function]}
          type="text"
          value="NFL"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="isEnabled"
      >
        Is Enabled
      </label>
      <div
        className="field"
      >
        <input
          checked={true}
          className="form-control"
          name="isEnabled"
          onChange={[Function]}
          type="checkbox"
          value="IsEnabled"
        />
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Saving...
    </button>
  </form>,
  <br />,
  <br />,
  <div
    className="row"
  >
    <div
      className="col-sm-3"
    >
      <div
        className="flex-column nav nav-pills"
        onKeyDown={[Function]}
        role="tablist"
      />
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="tab-content"
      />
    </div>
  </div>,
]
`;
